WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderTagHeaderView.kt

```diff
@@ -18,20 +23,64 @@ class ReaderTagHeaderView @JvmOverloads constructor(
     attrs: AttributeSet? = null,
     defStyleAttr: Int = 0
 ) : RelativeLayout(context, attrs, defStyleAttr) {
-    private val binding = ReaderTagHeaderViewBinding.inflate(LayoutInflater.from(context), this, true)
+    private var binding: ReaderTagBinding
 
     @Inject
     lateinit var uiHelpers: UiHelpers
 
+    @Inject
+    lateinit var readerImprovementsFeatureConfig: ReaderImprovementsFeatureConfig
+
+    @Inject
+    lateinit var localeProvider: LocaleProvider
+
     private var onFollowBtnClicked: (() -> Unit)? = null
 
     init {
         (context.applicationContext as WordPress).component().inject(this)
+        binding = if (readerImprovementsFeatureConfig.isEnabled()) {
+            val readerTagHeaderViewNewBinding =
+                ReaderTagHeaderViewNewBinding.inflate(LayoutInflater.from(context), this, true)
+            ReaderTagBinding.ImprovementsEnabled(
+                textTag = readerTagHeaderViewNewBinding.textTag,
+                followButton = readerTagHeaderViewNewBinding.followContainer.followButton,
+                textTagFollowCount = readerTagHeaderViewNewBinding.followContainer.textBlogFollowCount,
+            )
+        } else {
+            val readerTagHeaderViewBinding =
+                ReaderTagHeaderViewBinding.inflate(LayoutInflater.from(context), this, true)
+            ReaderTagBinding.ImprovementsDisabled(
+                textTag = readerTagHeaderViewBinding.textTag,
+                followButton = readerTagHeaderViewBinding.followButton,
+            )
+        }
         binding.followButton.setOnClickListener { onFollowBtnClicked?.invoke() }
     }
 
+    abstract class ReaderTagBinding {
+        abstract val textTag: MaterialTextView
+        abstract val followButton: ReaderFollowButton
+
+        data class ImprovementsDisabled(
+            override val textTag: MaterialTextView,
+            override val followButton: ReaderFollowButton
+        ) : ReaderTagBinding()
+
+        data class ImprovementsEnabled(
+            override val textTag: MaterialTextView,
+            override val followButton: ReaderFollowButton,
+            val textTagFollowCount: MaterialTextView,
+        ) : ReaderTagBinding()
+    }
+
     fun updateUi(uiState: ReaderTagHeaderUiState) = with(binding) {
+        (binding as? ReaderTagBinding.ImprovementsEnabled)?.textTagFollowCount?.gone()
+        // creative-writing -> Creative Writing
         textTag.text = uiState.title
+            .split("-")
+            .joinToString(separator = " ") {
+                it.replaceFirstChar { it.titlecase(localeProvider.getAppLocale()) }
+            }
```

Comment:
💡 Maybe this formatting should be done where we set the title instead, and not in the View.

https://github.com/wordpress-mobile/WordPress-Android/blob/183dca9bf0bcf0641e9f19e249b3bc7e49c22c54/WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java#L303